src/main/webapp/directory.html

```diff
@@ -0,0 +1,75 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8">
+    <title>Directory</title>
+    <link rel="stylesheet" href="style.css">
+    <script src="script.js"></script>
+  </head>
+  <body>
+    <div id="content">
+      <p align="right"><button onclick="location.href='addevent.html';">Add Event</button></p>
+
+      <h1>Happening in &lt;School Name&gt;</h1>
+
+      <form action="/filter" method="POST">
+        <table style="width:100%; text-align:left;">
+          <tr>
+            <th>Event type:</th>
+            <th>Subject:</th>
+            <th>Sort by:</th>
```

Comment:
For future reference: The sorting of items can be done in your JavaScript code and doesn't necessarily have to be a call to the backend.

Filtering should be a backend call because you need to query all of the correct items from datastore, but sorting can be done in you JavaScript since you will already have access to the items that you want to display.

Reply:
One thing that comes up is if we implement only loading some events at a time. Changing the parameter to sort by might change the events that should be displayed which would require fetching the events again. If we do that I think sorting would need to interact with the backend but alternatively we could just resort the events that are already displayed. I changed the HTML to correspond to the second option for now.

Reply:
I see what you mean. I think for now, we could just stick with requesting all events and storing them in your JavaScript code (an array/list for example). You can then sort them in this array, and display only a few at a time. I think this is the easiest for now. Since it is not an infinite scrolling scenario, and we won't have to fetch that many events, we can stick to just requesting them all and storing in JavaScript array instead of doing another HTTP request to get more.